### PR TITLE
fix(dynamite): recursively use for generic in TypeResult.dartType

### PR DESCRIPTION
**Pull request by Leptopoda**
Signed-off-by: Nikolas Rimikis <leptopoda@users.noreply.github.com>

### CODE REVIEW

**Comment by provokateurin** [comment]
Wrong PR title

File: packages/dynamite/dynamite/lib/src/models/type_result/enum.dart [hunk 1]
```diff
@@ -15,9 +15,16 @@ class TypeResultEnum extends TypeResult {
...
+        nullable: nullable,
+        isTypeDef: isTypeDef,
+      );
 
   @override
-  int get hashCode => className.hashCode + generics.hashCode + subType.hashCode;
+  bool operator ==(Object other) => other is TypeResultEnum && other.className == className && other.subType == subType;
+
+  @override
+  int get hashCode => className.hashCode + generics.hashCode + subType.hashCode + subType.hashCode;
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:29]:
```suggestion
  int get hashCode => className.hashCode + generics.hashCode + subType.hashCode;
```
Duplicate

File: packages/dynamite/dynamite/lib/src/models/type_result/enum.dart [hunk 2]
```diff
@@ -15,9 +15,16 @@ class TypeResultEnum extends TypeResult {
...
+  TypeResultEnum get dartType => TypeResultEnum(
+        className,
+        subType.dartType,
+        nullable: nullable,
+        isTypeDef: isTypeDef,
+      );
 
   @override
-  int get hashCode => className.hashCode + generics.hashCode + subType.hashCode;
+  bool operator ==(Object other) => other is TypeResultEnum && other.className == className && other.subType == subType;
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:26]: Why are the generics no longer compared?

**Review by provokateurin — CHANGES_REQUESTED** [review]
Why exactly is this needed or what was broken?

File: packages/dynamite/dynamite/lib/src/models/type_result/enum.dart (hunk 2 above)
Comment (Leptopoda) [reply packages/dynamite/dynamite/lib/src/models/type_result/enum.dart:26]: they shouldn't have been there from the beginning as an enum can not have any generics

**Comment by Leptopoda** [comment]
> Why exactly is this needed or what was broken?

Some code generation for enums was broken that I needed for #1881. The change I was trying to make was wrong and we will probably not end up picking it up. Nevertheless a dartType should always be a fully dartType (something jsonEncode can work with) so I'm still for this PR

**Review by provokateurin — APPROVED** [review]

**Pull request merged**